vscode-extension/src/openinvscode.ts

```diff
@@ -0,0 +1,77 @@
+import { Uri, commands, window } from 'vscode';
+import fetch from 'node-fetch';
+import { exec } from 'child_process';
+import FormData = require('form-data');
+import { chmod, open, writeFile } from 'fs/promises';
+
+export async function handleOpenInVSCode(uri: Uri) {
+    const queryParams = new URLSearchParams(uri.query);
+
+    if (queryParams.has('vm_id') && queryParams.has('token')) {
+        window.showInformationMessage('Setting up devbox');
+
+        // getting ssh keys
+        const response = await getVMInfo(queryParams.get('token'), queryParams.get('vm_id'));
+        const res = await response.json();
+        // TODO: remove debug
+        console.log("data:");
+        console.log(res);
+        // set ssh config
+        await setupSSHConfig(res['vm_id'], res['private_key']);
+        // connect to remote vm
+        connectToRemote(res['username'], res['vm_id']);
+    } else {
+        window.showErrorMessage('Error parsing information for remote environment.');
+    };
+}
+
+async function getVMInfo(token: string | null, vmId: string | null): Promise<any> {
+    // send post request to gateway to get vm info and ssh keys
+    const gatewayHost = 'https://api.devbox.sh/g/vm_info';
+    const data = new FormData();
+    data.append("vm_id", vmId);
+    const response = await fetch(gatewayHost, {
+        method: 'post',
+        body: data,
+        headers: {
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            'Authorization': `Bearer ${token}`
```

Comment:
`authorization` or `Authorization` doesn't work? Does it have to be quoted?

Reply:
I followed fetch's example from its own docs. I don't think it's necessary but I'll test

Reply:
followup: tried `authorization` without quotes and it worked. So will leave it as lowercase without quote `authorization`